modules/update-cvo-conditions.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding-update-service.adoc
+
+[id="update-cvo-conditions_{context}"]
+= Understanding CVO conditions
+
+The `ClusterVersion` object sets the conditions that describe the state of the cluster-version operator (CVO).
+
+Failing:: When `Failing` is set to True, the CVO fails to reconcile the cluster with the desired release image. In all cases, the impact on the cluster is that the dependent nodes in the manifest graph may not be reconciled. The graph may be flattened, in which case there are no dependent nodes.
++
+Most reconciliation errors result in `Failing=True`, even though `ClusterOperatorNotAvaliable` has special handling.
```

Comment:
This is stale since [4.7][1] ([not backported to 4.6][2]).  In the current 4.9 tip, we're fail-fast except for:

* [ClusterOperator resources that we fail to GET from the API-server][3] (currently one of a few cases that can create `ClusterOperator(s)NotAvailable`).
* ClusterOperator which are `Degraded=True`, which [get `ClusterOperatorDegraded` and a grace period before we go `Failing=True`][4].
* ClusterOperator with unexpected `status.versions`, which [get `ClusterOperatorUpdating`, and currently are never sufficient to go `Failing=True`][5].

We keep iterating in this space, so trying to pin down the reconciliation errors that do not trigger `Failing=True` in a given OpenShift version may not be worth the trouble.  Instead, maybe just point out that there are some cases where behavior is not happy enough to allow update progression, but not sad enough to warrant `Failing=True`, and maybe call out "waiting for `status.versions` to level in a ClusterOperator during updates" as a core example of that sort of thing.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1884334#c41
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1884464#c2
[3]: https://github.com/openshift/cluster-version-operator/blob/0ec39d9b2ab1feee8815d7b6b4bbe2db23daf847/pkg/cvo/internal/operatorstatus.go#L136-L145
[4]: https://github.com/openshift/cluster-version-operator/blob/0ec39d9b2ab1feee8815d7b6b4bbe2db23daf847/pkg/cvo/internal/operatorstatus.go#L206-L218
[5]: https://github.com/openshift/cluster-version-operator/blob/0ec39d9b2ab1feee8815d7b6b4bbe2db23daf847/pkg/cvo/internal/operatorstatus.go#L220-L230

Reply:
Closing this since content has drastically changed.